src/client/components/form-elements/ReviewCollapsiblePanel.jsx

```diff
@@ -1,62 +1,102 @@
 import React from 'react';
+import Scroll from 'react-scroll';
 import { connect } from 'react-redux';
 import _ from 'lodash';
 
-import ErrorableCheckbox from './ErrorableCheckbox';
-import { updateIncompleteStatus, updateVerifiedStatus, updateCompletedStatus } from '../../actions';
+import { ensureFieldsInitialized, updateIncompleteStatus, updateVerifiedStatus, updateCompletedStatus } from '../../actions';
+import * as validations from '../../utils/validations';
 
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
 
 /**
  * A component for the review section to validate information is correct.
  *
  * Required props
  */
 
-
 class ReviewCollapsiblePanel extends React.Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.handleSave = this.handleSave.bind(this);
+    this.handleNext = this.handleNext.bind(this);
     this.handleEdit = this.handleEdit.bind(this);
+    this.scrollToTop = this.scrollToTop.bind(this);
   }
 
   componentWillMount() {
     this.id = _.uniqueId();
   }
 
+  scrollToTop() {
+    scroller.scrollTo('topScrollReviewPanel', {
+      duration: 500,
+      delay: 2,
+      smooth: true,
+    });
+  }
+
   handleSave() {
     const currentPath = this.props.updatePath;
-    this.props.onUpdateSaveStatus(currentPath);
+    const formData = this.props.data;
+    const sectionFields = this.props.uiData.sections[currentPath].fields;
+
+    this.props.onFieldsInitialized(sectionFields);
+    if (validations.isValidSection(currentPath, formData)) {
+      this.props.onUpdateSaveStatus(currentPath);
+      this.scrollToTop();
+    }
+    this.scrollToTop();
+  }
+
+  handleNext() {
+    const currentPath = this.props.updatePath;
+    this.props.onUpdateVerifiedStatus(currentPath, true);
+    // TODO: find a better solution for this or a different implementation.
+    const delay = 100; // 0.1 second
```

Comment:
So my idea of using callbacks didn't work because the `onUpdateVerifiedStatus` completes before the component renders, which is why we need the delay (one step behind you). So I think what you have is the best option for now. There's not really a need to set a variable for the delay since we're only using it one time; I think it's fine to just pass the value in directly.